### PR TITLE
KAA-1399: KaaClientStateDelegate: crash, when method onStarted not implemented

### PR DESCRIPTION
**Pull request by vilimets**
@abohomol please review

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: client/client-multi/client-objective-c/KaaTests/KaaClientTest.m [hunk 1]
```diff
@@ -23,6 +23,38 @@
...
 
+
+@interface ClientStateDelegateFailure : NSObject
+
+- (void)onStartFailureWithException:(NSException *)exception;
+- (void)onPauseFailureWithException:(NSException *)exception;
+- (void)onResumeFailureWithException:(NSException *)exception;
+- (void)onStopFailureWithException:(NSException *)exception;
+
+
```
Comment (abohomol) [inline client/client-multi/client-objective-c/KaaTests/KaaClientTest.m]: Please, remove redundant lines

File: client/client-multi/client-objective-c/KaaTests/KaaClientTest.m [hunk 2]
```diff
@@ -115,6 +147,27 @@ - (void)testBasicLifeCycle {
     [verifyCount(self.delegate, times(1)) onStopped];
 }
 
+- (void)testKlientStateDelegateEmpty {
```
Comment (abohomol) [inline client/client-multi/client-objective-c/KaaTests/KaaClientTest.m]: Fix typo in word 'Klient'

File: client/client-multi/client-objective-c/KaaTests/KaaClientTest.m [hunk 3]
```diff
@@ -23,6 +23,38 @@
 #import <XCTest/XCTest.h>
 #import <Kaa/Kaa.h>
 
+
+@interface ClientStateDelegateFailure : NSObject
```
Comment (abohomol) [inline client/client-multi/client-objective-c/KaaTests/KaaClientTest.m]: I'd rather put there a name that will give an idea of possible issue. (e.g. ClientStateDelegateWithoutImplementedMethods)

**Comment by abohomol** [comment]
LGTM

**Comment by vadimol** [comment]
@kaa-leeroy add to whitelist

**Comment by rasendubi** [comment]
@kaa-leeroy add to whitelist

**Review by vadimol — APPROVED** [review]

**Pull request merged**